server/src/test/java/org/apache/cassandra/sidecar/handlers/OpenApiHandlerTest.java

```diff
@@ -0,0 +1,310 @@
+                  assertThat(response.statusCode()).isEqualTo(HttpResponseStatus.OK.code());
+                  assertThat(response.getHeader("Content-Type")).isEqualTo("application/yaml");
+                  assertThat(response.bodyAsString()).isEqualTo(TEST_YAML_CONTENT);
+                  testContext.completeNow();
+              })));
+    }
+    
+    @Test
+    void testDefaultToJsonWhenNoFormatSpecified(VertxTestContext testContext)
+    {
+        client.get(server.actualPort(), "localhost", "/openapi")
+              .timeout(5000)
+              .send()
+              .onComplete(testContext.succeeding(response -> testContext.verify(() -> {
+                  assertThat(response.statusCode()).isEqualTo(HttpResponseStatus.OK.code());
+                  assertThat(response.getHeader("Content-Type")).isEqualTo("application/json");
+                  assertThat(response.bodyAsString()).isEqualTo(TEST_JSON_CONTENT);
+                  testContext.completeNow();
+              })));
+    }
+
+    @Test
+    void testIsYamlRequestDetection()
+    {
+        RoutingContext context = mock(RoutingContext.class);
+        io.vertx.core.http.HttpServerRequest request = mock(io.vertx.core.http.HttpServerRequest.class);
+        when(context.request()).thenReturn(request);
+        
+        // Test path-based detection
+        when(request.path()).thenReturn("/openapi.yaml");
+        assertThat(handler.testIsYamlRequest(context)).isTrue();
+        
+        when(request.path()).thenReturn("/openapi.json");
+        assertThat(handler.testIsYamlRequest(context)).isFalse();
+        
+        // Test accept header detection
+        when(request.path()).thenReturn("/openapi");
+        when(request.getHeader("Accept")).thenReturn("application/yaml");
+        assertThat(handler.testIsYamlRequest(context)).isTrue();
+        
+        when(request.getHeader("Accept")).thenReturn("application/json");
+        assertThat(handler.testIsYamlRequest(context)).isFalse();
+        
+        when(request.getHeader("Accept")).thenReturn("text/html, application/yaml");
+        assertThat(handler.testIsYamlRequest(context)).isTrue();
+        
+        when(request.getHeader("Accept")).thenReturn(null);
+        assertThat(handler.testIsYamlRequest(context)).isFalse();
+    }
+    
+    private void deleteRecursively(Path path) throws IOException
+    {
+        if (Files.isDirectory(path))
+        {
+            Files.list(path).forEach(child -> {
+                try
+                {
+                    deleteRecursively(child);
+                }
+                catch (IOException e)
+                {
+                    // Best effort cleanup
+                }
+            });
+        }
+        Files.deleteIfExists(path);
+    }
+    
+    /**
+     * Test subclass that overrides resource loading for predictable testing
+     */
+    private static class TestOpenApiHandler extends OpenApiHandler
```

Comment:
hmm... looks like some tests are testing code in test scope... Can you write the test similar to `org.apache.cassandra.sidecar.handlers.cassandra.NodeSettingsHandlerTest`?